packages/shared/lib/core/utils/object.ts

```diff
@@ -10,30 +10,33 @@
 export function migrateObjects<T>(oldObj: T, newObj: T): T {
     /* eslint-disable @typescript-eslint/no-explicit-any */
     const _helper = (curObj: any, oldObj: any, newObj: any): any => {
-        // Iterate through each key of a new object...
-        for (const k in newObj) {
+        // Iterate through each key of a new AND old objects...
+        for (const key in { ...oldObj, ...newObj }) {
             // If key also exists in old object then...
-            if (k in oldObj) {
+            if (key in oldObj) {
                 // If corresponding value in the new object is also an object and is not undefined or null then...
-                if (typeof newObj[k] === 'object' && newObj[k] !== undefined && newObj[k] !== null) {
+                if (typeof newObj[key] === 'object' && newObj[key] !== undefined && newObj[key] !== null) {
                     // If corresponding value is actually an array (b/c an "Array" is an object in JS) then...
-                    if (Array.isArray(newObj[k])) {
+                    if (Array.isArray(newObj[key])) {
                         // Create key-value pair with array from new object or old object if it already exists
-                        curObj[k] = Array.isArray(oldObj[k]) ? oldObj[k] : newObj[k]
+                        curObj[key] = Array.isArray(oldObj[key]) ? oldObj[key] : newObj[key]
+                        // Else if corresponding value is actually a date
+                    } else if (typeof newObj[key]['getMonth'] === 'function') {
+                        curObj[key] = newObj[key]
```

Comment:
if we have a date in the `oldObj`, it gets always overridden with the `newObj`s date